src/tests/ftest/osa/osa_online_reintegration.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2020 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+import time
+import random
+import uuid
+import threading
+
+from itertools import product
+from avocado import fail_on
+from apricot import TestWithServers
+from test_utils_pool import TestPool
+from ior_utils import IorCommand
+from job_manager_utils import Mpirun
+from write_host_file import write_host_file
+from command_utils import CommandFailure
+from mpio_utils import MpioUtils
+
+try:
+    # python 3.x
+    import queue as queue
+except ImportError:
+    # python 2.7
+    import Queue as queue
+
+
+class OSAOnlineReintegration(TestWithServers):
```

Comment:
You could inherit IorTestBase to cut down most of the ior related code. It will take care of loading ior properties from yaml and ready to use self.ior object.

Reply:
Agree. Myself and Samir had issues using the IORTestbase as is... The reason is the code tries to create multiple pools and multiple IOR threads. So we adopted this approach in NVMe testing which I tried to use it here.